python_modules/libraries/dagster-teradata/dagster_teradata/resources.py

```diff
@@ -0,0 +1,464 @@
+            :param aws_conn_id: The Airflow AWS connection used for AWS credentials.
+            :param teradata_conn_id: The connection ID used to connect to Teradata
+                :ref:`Teradata connection <howto/connection:Teradata>`.
+            :param teradata_authorization_name: The name of Teradata Authorization Database Object,
+                is used to control who can access an S3 object store.
+                Refer to
+                https://docs.teradata.com/r/Enterprise_IntelliFlex_VMware/Teradata-VantageTM-Native-Object-Store-Getting-Started-Guide-17.20/Setting-Up-Access/Controlling-Foreign-Table-Access-with-an-AUTHORIZATION-Object
+
+        Examples:
+            .. code-block:: python
+
+                import pandas as pd
+                import pyarrow as pa
+                import pyarrow.parquet as pq
+
+                @op
+                def write_parquet_file(teradata: TeradataResource):
+                    df = pd.DataFrame({"one": [1, 2, 3], "ten": [11, 12, 13]})
+                    table = pa.Table.from_pandas(df)
+                    pq.write_table(table, "example.parquet')
+                    teradata.load_table_from_local_parquet(
+                        src="example.parquet",
+                        table="MY_TABLE"
+                    )
+
+        """
+        credentials_part = "ACCESS_ID= '' ACCESS_KEY= ''"
+
+        if not public_bucket:
+            # Accessing data directly from the S3 bucket and creating permanent table inside the database
+            if teradata_authorization_name:
+                credentials_part = f"AUTHORIZATION={teradata_authorization_name}"
+            else:
+                access_key = s3.aws_access_key_id
+                access_secret = s3.aws_secret_access_key
+                token = s3.aws_session_token
+                credentials_part = f"ACCESS_ID= '{access_key}' ACCESS_KEY= '{access_secret}'"
+                if token:
+                    credentials_part = credentials_part + f" SESSION_TOKEN = '{token}'"
+
+        sql = dedent(f"""
+                    CREATE MULTISET TABLE {teradata_table} AS
+                    (
+                        SELECT * FROM (
+                            LOCATION = '{s3_source_key}'
+                            {credentials_part}
+                        ) AS d
+                    ) WITH DATA
+                    """).rstrip()
+
+        self.execute_query(sql)
+
+    @public
+    def azure_blob_to_teradata(
+        self,
+        azure_client_id: str,
+        azure_client_secret: str,
+        blob_source_key: str,
+        teradata_table: str,
+        public_bucket: bool = False,
+        teradata_authorization_name: str = "",
+    ):
+        """Loads CSV, JSON, and Parquet format data from Azure Blob Storage to Teradata.
+
+        Args:
+            :param blob_source_key: The URI format specifying the location of the Azure blob object store.
```

Comment:
documentation missed for azure_client_id and azure_client_secret. Rearrange documentation of fields as per method parameters defined order

Reply:
I will add that